clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -1248,6 +1248,7 @@ public void closeShouldBeIdempotent() {
         KafkaConsumer<byte[], byte[]> consumer = newConsumer();
         consumer.close();
         consumer.close();
+        consumer.close();
```

Comment:
The third call didn't expose the bug, but it seemed like a good thing to have. It ensures that the second call to `close` (which is the first time we execute the "already closed" path) does not alter the state in any way which prevents additional calls.